modules/instances/main.tf

```diff
@@ -47,6 +47,27 @@ resource "aws_key_pair" "pe_adm" {
   tags       = local.tags
 }
 
+# Additional internalDNS depends on instances to be deployed to determine the
+# value of private_dns, instance resource ignore content of this tag
+resource "aws_ec2_tag" "server-dns" {
+  count       = length(aws_instance.server[*])
+  resource_id = aws_instance.server[count.index].id
+  key         = "internalDNS"
+  value       = var.domain_name == null ? aws_instance.server[count.index].private_dns : "pe-server-${count.index}-${var.id}.${var.domain_name}"
```

Comment:
Is it possible in Terraform to refactor this expression to be used in one place instead of 3? For instance, where the variable is defined?

Reply:
Yes. New commits pushed.